metricbeat/module/haproxy/haproxy.go

```diff
@@ -251,18 +253,21 @@ func (p *unixProto) run(cmd string) (*bytes.Buffer, error) {
 
 	conn, err := net.Dial(p.Network, p.Address)
 	if err != nil {
-		return response, err
+		return response, errors.Wrap(err, "Error connecting")
 	}
 	defer conn.Close()
 
 	_, err = conn.Write([]byte(cmd + "\n"))
 	if err != nil {
-		return response, err
+		return response, errors.Wrap(err, "Error writing to connection")
 	}
 
-	_, err = io.Copy(response, conn)
+	recv, err := io.Copy(response, conn)
 	if err != nil {
-		return response, err
+		return response, errors.Wrap(err, "error reading response")
+	}
+	if recv == 0 {
+		return response, errors.New("Got empty response from HAProxy")
```

Comment:
This error messages should be lowercased.
```suggestion
		return response, errors.New("got empty response from HAProxy")
```

Same for other error messages added here.